publisher/internal/analytics/ethereum/process.go

```diff
@@ -22,65 +12,34 @@ func (a *Analytics) ProcessMessage(msg analytics.Message, send analytics.Sender)
 	if !hasTopics(eLog) {
 		return nil
 	}
-	// 1) All events are put into cache
-	a.addLogToTxCache(eLog)
 
-	if !a.isMintEvent(eLog) || (a.isMintEvent(eLog) && !isUniswapPositionsNFT(eLog.Data)) {
-		return nil
-	}
+	a.addLogToTxCache(eLog) //All events are put into cache
 
-	// 2) Mint event is found
-	var liquidityEntry LiquidityAddition
-	txEventsFromCache, _ := a.eventLogCache.Get(eLog.TransactionHash)
-	for _, evLog := range txEventsFromCache.([]EventLog) { // Go through all events of this transaction
-		if reflect.DeepEqual(evLog, eLog) { // This allows to correctly process many Mint events in one transaction
-			break
-		}
-		if a.isAvoidEvent(evLog) {
-			liquidityEntry = LiquidityAddition{} // Reset gathered records
-			continue
-		}
-		if a.isTransferEvent(evLog) { // 3) Searching for relevant "Transfer" event(s)
-			liquidityEntry = a.handleLiquidityTransfer(evLog, liquidityEntry) // 4) Decoding / expanding "Transfer" events
-		}
-	}
-	liquidityEntry = a.checkAndUpdateMissingToken(eLog, liquidityEntry)         // 5) Adding missing token if only 1 token transfer was made
-	liquidityEntry = a.includeTokenPrices(liquidityEntry)                       // 6) Getting token prices
-	liquidityEntry, doPublish := a.finalizeLiquidityEntry(eLog, liquidityEntry) // 7) Save Liquidity Entry and Liquidity Pool
-
-	if !doPublish {
+	var operation Operation
+	switch {
+	case a.isBurnEvent(eLog):
+		operation = Removal{}
+	case a.isMintEvent(eLog):
+		operation = Addition{}
+	default:
 		return nil
 	}
-	message, err := makeMessage(liquidityEntry)
+
+	var err error
+	operation, err = operation.ExtractFromEventLogs(eLog, a)
 	if err != nil {
-		log.Println(err)
-		return err
+		log.Printf("%s\n", err)
+		return nil
+		//return err //TODO: if error is returned - whole service is stopped
 	}
-	streamName := strings.ToLower(fmt.Sprintf("%s.%s", liquidityEntry.Token0Symbol, liquidityEntry.Token1Symbol))
-	err = send(message, streamName)
-	return err
-}
-
-func makeMessage(liqAdd LiquidityAddition) ([]byte, error) {
 
-	additionMessage := types.AdditionMessage{
-		Timestamp:         time.Now().UTC(), // Add a timestamp field
-		Address:           liqAdd.LPoolAddress,
-		LowerTokenRatio:   liqAdd.LowerRatio,
-		CurrentTokenRatio: liqAdd.CurrentRatio,
-		UpperTokenRatio:   liqAdd.UpperRatio,
-		ValueAddedUSD:     liqAdd.ValueAdded,
-		Pair: [2]types.TokenMessage{
-			{Symbol: liqAdd.Token0Symbol, Amount: liqAdd.Token0Amount, Price: liqAdd.Token0PriceUsd},
-			{Symbol: liqAdd.Token1Symbol, Amount: liqAdd.Token1Amount, Price: liqAdd.Token1PriceUsd},
-		},
-		TxHash: liqAdd.TxHash,
+	if !operation.CanPublish() {
+		return nil
 	}
 
-	additionJson, err := json.Marshal(&additionMessage)
-	if err != nil {
-		return nil, fmt.Errorf("Error marshalling Liquidity Addition object into a json message: %s", err)
-	}
+	log.Println(eLog.TransactionHash)
```

Comment:
This and other places with logs: Logs should be descriptive and easily understood. Containing a short prefix description of the log contents at minimum.